### PR TITLE
FAI-540 - deterministic LIME sample size

### PR DESCRIPTION
**Pull request by tteofili**
See https://issues.redhat.com/browse/FAI-540

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Comment by tteofili** [comment]
Jenkins retest this please

File: explainability/explainability-core/src/main/java/org/kie/kogito/explainability/local/lime/LimeExplainer.java [hunk 1]
```diff
@@ -90,13 +90,19 @@ public LimeConfig getLimeConfig() {
         }
         List<Output> actualOutputs = prediction.getOutput().getOutputs();
 
+        int noOfSamples = limeConfig.getNoOfSamples();
+
+        if (noOfSamples <= 0) {
+            noOfSamples = (int) Math.pow(2, linearizedTargetInputFeatures.size());
```
Comment (danielezonca) [inline explainability/explainability-core/src/main/java/org/kie/kogito/explainability/local/lime/LimeExplainer.java:96]: Can you add a test to trigger this path?
Can you also add a log message when this happen?

Comment (tteofili) [reply explainability/explainability-core/src/main/java/org/kie/kogito/explainability/local/lime/LimeExplainer.java:96]: sure

**Review by danielezonca — APPROVED** [review]

**Review by ruivieira — APPROVED** [review]
LGTM, thanks @tteofili

**Comment by tteofili** [comment]
Jenkins retest this please

**Review by jiripetrlik — APPROVED** [review]

**Comment by danielezonca** [comment]
Jenkins retest this please

**Pull request merged**